src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -331,12 +330,29 @@ class AppsController(
               complete((StatusCodes.OK, List(Headers.`Marathon-Deployment-Id`(plan.id)), DeploymentResult(plan.id, plan.version.toOffsetDateTime)))
             }
           case TaskKillingMode.Wipe =>
-            onSuccess(taskKiller.kill(appId, findToKill, wipe = true)) { instances =>
-              complete(raml.InstanceList(instances.map(_.toRaml)))
+            val killInstances = async {
+              val instances = await(taskKiller.kill(appId, findToKill, wipe = true))
+              // TODO: Does it make sense to set health for a killed task?
```

Comment:
What do you think, @meichstedt?

Reply:
@jeschkies would it be useful to see why a task was killed? (failed health check?)

I'm inlined towards no, not useful to include health check with killed tasks.

Reply:
@jeschkies AFAICT the result is meant to provide the information which tasks have been killed. I don't think we need to add health information here and in other locations where we kill tasks via the API, and I think the related logic should be removed.

@timcharper `TaskKiller` will set `KillReason.KillingTasksViaApi`. If the task is killed because it failed health checks too often, `KillReason.FailedHealthChecks` will be logged. IMO, when tasks are killed via the AppsController, `KillingTasksViaApi` is always correct.

Reply:
Sounds like we can omit the health check information here, then.